instructionAPI/src/ArchSpecificFormatters.C

```diff
@@ -149,6 +149,61 @@ std::string ArmFormatter::formatBinaryFunc(std::string left, std::string func, s
         return left + " " + func + " " + right;
 }
 
+///////// Formatter for AMDGPU
+
+AmdgpuFormatter::AmdgpuFormatter() {
+    binaryFuncModifier["<<"] = "lsl";
+}
+
+std::string AmdgpuFormatter::formatImmediate(std::string evalString) {
+    return "0x" + evalString;
+}
+
+std::string AmdgpuFormatter::formatRegister(std::string regName) {
+    std::string ret = regName;
+    std::transform(ret.begin(), ret.end(), ret.begin(), ::toupper);
```

Comment:
As of C++11, having the destination iterator be part of the input range is undefined behavior. Changing this to a simple range-based loop is the easy fix.

    for(auto &c : ret) ::toupper(c);

Also, there's no need to make a copy of the parameter since it's passed by value.

Reply:
Fixed